autoload/neomake/makers/ft/text.vim

```diff
@@ -27,3 +34,51 @@ function! neomake#makers#ft#text#writegood() abort
                 \ 'postprocess': function('neomake#makers#ft#text#PostprocessWritegood'),
                 \ }
 endfunction
+
+let s:languagetool_fallback_language = 'auto'
+" See http://wiki.languagetool.org/public-http-api for a public instance. Use:
+"   :let g:neomake_text_languagetool_server = 'https://languagetool.org/api'
+let s:languagetool_fallback_server = 'http://localhost:8081'
+
+let s:languagetool_maker = {}
+let s:languagetool_maker.exe = expand('<sfile>:p:h', 1).s:slash.'text'.s:slash.'languagetool.py'
+let s:languagetool_maker.append_file = 1
+function! s:languagetool_maker.InitForJob(_jobinfo) abort
+    let l:args = []
+    " Mandatory arguments
+    let l:server = s:getVar('neomake_text_languagetool_server', s:languagetool_fallback_server)
+    let l:language = s:getVar('neomake_text_languagetool_language',
+                \ get(split(&spelllang, ','), 0, s:languagetool_fallback_language) )
+    " Optional Arguments
+    let motherTongue = s:getVar('neomake_text_languagetool_curl_motherTongue', '')
+    if !empty(motherTongue)
+        let args += ['--motherTongue', motherTongue]
+    endif
+    let preferredVariants = s:getVar('neomake_text_languagetool_curl_preferredVariants', '')
+    if !empty(l:preferredVariants) && l:language ==# 'auto'
+        for var in l:preferredVariants
+            let args += ['--preferredVariants', var]
+        endfor
+    endif
+    let l:args += [l:server, l:language]
+    let self.args = l:args
+endfunction
```

Comment:
Instead of extra args this should could just rely on `g:/b:neomake_text_languagetool_args`, and print an error if not set?!

Reply:
The args are composed of the rest of the settings. How would I compose them, in the same function?

Reply:
Is the `neomake#utils#GetSetting` still the "recommended" way of accessing the settings? How can I use it, it has a bunch of required arguments I wouldn't want to hardcode.

Reply:
Yeah, `neomake#utils#GetSetting` is still the "recommended" way, since it handles both old-style and new-style settings.
The newer API should be saner, but then you would have to require new-style settings.
You should look at existing usages of it - typically you are not hard-coding settings, but e.g. only passing in the maker.

> The args are composed of the rest of the settings. How would I compose them, in the same function?

I got the impression that there are just multiple args that are all added up, i.e. you would just require `g:neomake_text_languagetool_args` to be set - no defaults there.

This gets automatically turned into arguments for the maker - i.e. there is only one required setting and that is `…_args`.  You should probably just check in InitForJob if `args` is empty, and then suggest that `g:/b:neomake_text_languagetool_args` should be configured.